Sources/MapboxNavigation/NavigationMapView+BuildingHighlighting.swift

```diff
@@ -23,17 +23,20 @@ extension NavigationMapView {
                                    in3D extrudesBuildings: Bool = true,
                                    extrudeAll: Bool = false,
                                    completion: ((_ foundAllBuildings: Bool) -> Void)? = nil) {
-        var foundBuildingIds = Set<Int64>()
+        highlightBuildings = highlightBuildings.filter{ coordinates.contains($0.key) }
         let group = DispatchGroup()
         let identifiers = mapView.mapboxMap.style.allLayerIdentifiers
             .compactMap({ $0.id })
             .filter({ $0.contains("building") })
         let layerPosition = identifiers.last.map { LayerPosition.above($0) }
         
-        coordinates.forEach {
-            group.enter()
+        for coordinate in coordinates {
+            let screenCoordinate = mapView.mapboxMap.point(for: coordinate)
+            if screenCoordinate == CGPoint(x: -1.0, y: -1.0) {
```

Comment:
If the coordinate is out of map boundary, don't do the map query, because it will return a false result.

Reply:
Because the `CGPoint(x: -1.0, y: -1.0)` is the value the `MapView.mapboxMap.point(for:)` will return if the coordinate is out of map bounds.

Reply:
I propose just to verify whether coordinate is out of map view bounds, if so - simply do not perform querying, and do not add anything to `foundBuildingIds`. At the end users will be notified that not all buildings at specific coordinates were highlighted and member variable in `NavigationMapView` will not be needed.

Reply:
Because the query is done inside the `mapView`, we only need to check whether the coordinate is outside the `mapView` bound. And that's the value the `mapView` will return if the coordinate is out of `mapView` bound.